mode/smalltalk/smalltalk.js

```diff
@@ -120,6 +122,17 @@ CodeMirror.defineMode('smalltalk', function(config) {
 
     if (aChar === '|') {
       token.context = context.parent;
+      var str = stream.string;
```

Comment:
Could whatever you're trying to do here be expressed by matching a regexp on the stream at this point? Munging the whole line without regard for where in the line we currently are doesn't seem like a good idea.